exams/tasks.py

```diff
@@ -2,56 +2,128 @@
 Tasks for exams
 """
 from datetime import datetime
+import logging
 import tempfile
 
+from django.conf import settings
 from django.db import transaction
 import pytz
 
-from exams.models import ExamProfile
-from exams.pearson import (
-    write_cdd_file,
-    upload_tsv,
+from exams.exceptions import RetryableSFTPException
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
 )
+from exams import pearson
 from micromasters.celery import async
 
+PEARSON_CDD_FILE_PREFIX = "cdd-%Y%m%d%H_"
+PEARSON_EAD_FILE_PREFIX = "ead-%Y%m%d%H_"
 
-@async.task
-def export_exam_profiles():
+PEARSON_FILE_EXTENSION = ".dat"
+
+PEARSON_FILE_ENCODING = "utf-8"
+
+log = logging.getLogger(__name__)
+
+
+def _backoff(retries):
+    """
+    Exponential backoff for retried tasks
+    """
+    return settings.EXAMS_SFTP_BACKOFF_BASE ** retries
+
+
+@async.task(bind=True, max_retries=3)
+def export_exam_profiles(self):
```

Comment:
@giocalitri I added some exception handling / logging here. In the case of certain exceptions that are retryable (`SSHException`, `ConnectionException`), those are re-reraised as `RetryableSFTPException` in `exams.pearson.upload_tsv` and trigger a task retry with exponential backoff. The default factor base of 5 will means these 3 reitries will spread over 2.5 minutes.

Otherwise, other exceptions will get logged and the task exits (and acks). I think that's acceptable because this task will eventually get run via celerybeat and the profiles that didn't get synced due to a previous error will sync on the next run (assuming the issue was either transient or resolved by us in the interim).